lib/Basic/UUID.cpp

```diff
@@ -15,40 +15,126 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <uuid/uuid.h>
 #include "swift/Basic/UUID.h"
 
+// WIN32 doesn't natively support <uuid/uuid.h>. Instead, we use Win32 APIs
+// class
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <rpc.h>
+#include <string>
+
+// Windows has a different layout of UUID than libuuid. However, they are the same
+// size, as mandated by the UUID specification (RFC 4122). This means that we can
+// convert the memory of a 128 bit unsigned char array to a ::UUID.
+// We also can't expose <rpc.h> in any headers, as it imports <windows.h> which
+// defines macros that conflict with many definitions (e.g. ERROR, std::max)
+union Conversion {
```

Comment:
I had no idea you can just memcpy like that - you learn something new everyday:

This works, and the file builds, so we're good:
```
#include <rpc.h>

int main()
{
	::UUID uuid1;
	unsigned char data[16] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16 };	
	memcpy(&uuid1, data, 16);

	// uuid1 = {04030201-0605-0807-090A-0B0C0D0E0F10}

    return 0;
}
```